### DISCUSSION TITLE
Avoid accidentally quitting

### DISCUSSION THREAD

**Discussion by reagle** [saulpw/visidata#1468] [Q&A] [answered]
Sometimes I'm a couple sheets down in my analysis and want to bounce back up to the top and accidentally overshoot and quit. What I'm thinking now:

1. Look at the sheet number and don't quit on sheet "1". (Kind of works, unless working with multiple top level sheets.)
2. Always guard the top sheets?

Has anyone considered something a bit more hierarchical? This could be a visual representation in Sheets View. Or some sort of "cookie trail." I was also thinking of naming a Frequency table off sheet 2 as 2.a or 2.1, but that would render the `alt` shortcut keys useless...

**Comment by anjakefala** [comment, marked as answer]
In case you do not know about it, the `--quitguard` option will confirm the quitting of a modified sheet.

**Reply by anjakefala** [reply]
If you always want `--quitguard` on, I would add `options.quitguard = True` to your `~/.visidatarc`. Please do open an issue if it does not behave in a way that you expect. There was a lot of love that users have put into refining its design: https://github.com/saulpw/visidata/issues?q=is%3Aissue+is%3Aclosed+quitguard

**Reply by reagle** [reply]
I did not! I only learned this was called "guard" after much toe-stubbing while looking for a way to "protect" sheets. Thank you.

**Comment by saulpw** [comment]
Hi @reagle, a hierarchical view of sheets is an interesting idea.  We have the `source` field on all sheets so it could be figured out.  I've been interested in making a little sheets panel that could show the sheets stack, maybe a tree view would work there.

I'm thinking about allowing `quitguard` to be set to other options, like `"source"` so that only the source sheets are guarded, as you suggest.
